spec/e2e/enum_spec.rb

```diff
@@ -68,6 +82,31 @@
       expect(StoredFile.as(:f).where(id: file.id).pluck('f.flag')).to eq([nil])
       expect(file.reload.flag).to eq(nil)
     end
+
+    it "raises error if value doesn't match an enum key" do
+      file = StoredFile.new
+      file.type = :audio
+      expect { file.save! }.to raise_error Neo4j::Shared::Enum::InvalidEnumValueError
+    end
+
+    it 'respects local _case_sensitive option' do
+      file = StoredFile.new
+      file.type_format = :png
+      expect { file.save! }.to raise_error(Neo4j::Shared::Enum::InvalidEnumValueError)
+
+      file.type_format = :Png
+      file.save!
+      expect(StoredFile.as(:f).pluck('f.type_format')).to eq([1])
+      expect(file.reload.type_format).to eq(:Png)
+    end
+
+    it 'respects global _case_sensitive = false default' do
+      file = StoredFile.new
+      file.type = :VIdeO
+      file.save!
+      expect(StoredFile.as(:f).pluck('f.type')).to eq([2])
+      expect(file.reload.type).to eq(:video)
+    end
```

Comment:
You could test the configuration by doing:

```ruby
context 'global enums_case_sensitive config is set to true' do
  let_config(:enums_case_sensitive) { true }
  # spec here testing, for example, that you can't set `type` to `VIdeO`
end
```

We define the `let_config` helper here:

https://github.com/neo4jrb/neo4j/blob/master/spec/spec_helper.rb#L135

Reply:
Oh thats cool